src/vs/workbench/contrib/notebook/browser/contrib/statusBar/contributedStatusBarItemController.ts

```diff
@@ -35,12 +35,17 @@ export class ContributedStatusBarItemController extends Disposable implements IN
 	}
 
 	private _updateEverything(): void {
-		this._visibleCells.forEach(cell => cell.dispose());
-		this._visibleCells.clear();
-		this._updateVisibleCells({ added: this._observer.visibleCells, removed: [] });
+		const newCells = this._observer.visibleCells.filter(cell => !this._visibleCells.has(cell.handle));
+		this._visibleCells.forEach(helper => helper.update());
+		const visibleCellHandles = new Set(this._observer.visibleCells.map(item => item.handle));
+		const removedCells = Array.from(this._visibleCells.keys()).filter(handle => !visibleCellHandles.has(handle));
+		this._updateVisibleCells({ added: newCells, removed: removedCells.map(handle => ({ handle })) });
 	}
 
-	private _updateVisibleCells(e: ICellVisibilityChangeEvent): void {
+	private _updateVisibleCells(e: {
```

Comment:
Did you mean to change this?

Reply:
Yes, because inside `_updateEverything` we don't have the `CellViewModel` in the `_visibleCells`, all we have is the handle information.
Hence trimmed the  interface to only what we need (we only need the handle & that's all we have when calling from `_updateEverything`